cpptest.cc

```
@@ -0,0 +1,23 @@
+
+int main()
+{
+  auto res = add(1, 2);
+  cout<<"res of add(1,2) is "<<res<<endl;
+  
+  Foo *foo = new Foo;
+  if (foo == nullptr) 
+  {
+    cout<<"Fail to allocate memory for Foo object!"<endl;
+  }
+}
```

Cppcheck (2.10-2)

[error] Memory leak

(memleak)